c++/src/kj/compat/http.c++

```diff
@@ -2177,10 +2177,14 @@ private:
     // is empty, then they make the write directly, using `writeInProgress` to detect and block
     // concurrent writes.
 
-    writeQueue = writeQueue.then([this,content=kj::mv(content)]() mutable {
-      auto promise = inner.write(content.begin(), content.size());
-      return promise.attach(kj::mv(content));
-    });
+    constexpr auto doWrite = [](Promise<void> writeQueue,
+                                AsyncOutputStream& inner,
+                                String content) -> Promise<void> {
+      co_await writeQueue;
+      co_await inner.write(content.begin(), content.size());
+    };
+
+    writeQueue = doWrite(kj::mv(writeQueue), inner, kj::mv(content));
```

Comment:
I feel like this function got more complicated TBH. Maybe it was better the way it was.

Reply:
Works for me.